lib/ReactViews/Clipboard.tsx

```diff
@@ -24,57 +23,51 @@ interface ClipboardProps {
 }
 
 const Clipboard: React.FC<ClipboardProps> = (props) => {
-  const { id, source, theme, rounded } = props;
+  const { id, source, theme, rounded, text, onCopy } = props;
   const { t } = useTranslation();
   const [status, setStatus] = useState<CopyStatus>(
     CopyStatus.NotCopiedOrWaiting
   );
-  useEffect(() => {
-    // Setup clipboard.js and show a tooltip on copy success or error for 3s
-    const clipboardBtn = new clipboard(`.btn-copy-${id}`);
-    let timerId: ReturnType<typeof setTimeout> | null = null;
-    function removeTimeout() {
-      if (timerId !== null) {
-        clearTimeout(timerId);
-        timerId = null;
+
+  const handleCopy = async () => {
+    try {
+      if (text) {
+        await navigator.clipboard.writeText(text);
+        setStatus(CopyStatus.Success);
+        if (onCopy) onCopy(text);
```

Comment:
If onCopy throws an exception, the status will go from `<initial state>` to `CopyStatus.Success` to `CopyStatus.Error` in a short while. Does it matter that some parts might observe the success and other parts might observe the error, or are all three of these state updates batched together so consistency is guaranteed?